signature/policy_eval_signedby.go

```diff
@@ -70,6 +71,13 @@ func (pr *prSignedBy) isSignatureAuthorAccepted(image types.UnparsedImage, sig [
 			return PolicyRequirementError(fmt.Sprintf("Signature by key %s is not accepted", keyIdentity))
 		},
 		validateSignedDockerReference: func(ref string) error {
+			r := image.Reference().DockerReference()
+			if _, isCanonical := r.(reference.Canonical); isCanonical || reference.IsNameOnly(r) {
+				if !pr.UntaggedReference.matchesDockerReference(image, ref) {
```

Comment:
`matchRepository`, `exactRepository` and `exactTag` should all work naturally (the tag either doesn’t matter or doesn’t come from the reference).

The default `matchExact`… good point. Back to the drawing board? (I guess it is fairly clear that we would accept any tag in that case, but the `acceptableSignedIdentity` naming breaks down.)